### ISSUE TITLE
Conflict with Immersive Petroleum

### ISSUE THREAD

**Issue by Xiaminou** [desht/pnc-repressurized#30]
I just added Immersive Petroleum and now the server crash on world load.
It's a conflict about the oil world generation.
I did try the solution given by the crash reports but either I've done it wrong or it doesn't work.

Crash Report: [crash-2017-11-10_02.58.55-server.txt](https://github.com/desht/pnc-repressurized/files/1459823/crash-2017-11-10_02.58.55-server.txt)

Minecraft: 1.12.2
Forge: 2531
Mods: [list.txt](https://github.com/desht/pnc-repressurized/files/1459829/list.txt)

**Comment by Zixxl** [comment]
The solution in the crash report does not make sense. You could try:
`[{"modId": "immersivepetroleum","deps": [{"type":"after","target":"pneumaticcraft"}]}]`

**Comment by desht** [comment]
Yeah, you're right.  It doesn't make much sense.  I'll see if I can get IP added to my dev instance today and have a play, but it might not happen until Monday...

I definitely do want oil compatibility between IP and PNC:R so count on this one getting fixed :)

**Comment by Zixxl** [comment]
I tested an instance with both mods installed and couldn't reproduce the crash. But without the injectedDependencies.json I couldn't properly load my saved world (some pnc stuff missing). After I created the file and added `[{"modId": "immersivepetroleum","deps": [{"type":"after","target":"pneumaticcraft"}]}]` everything seems to load fine.

**Comment by desht** [comment]
I've actually just made a change which *appears* to resolve the problem without the need for a separate dependency injection.  I've tested on a dedicated server instance and it starts up fine.

Note that IP actually replaces three PNC:R fluids (oil, diesel and lubricant).  With this change, PNC:R behaves better if it finds one of its fluids already registered; however, if the other mod hasn't registered a block for the fluid, then PNC:R won't either.  With my testing, all three fluids appears as the IP fluid. Compatibility looks good - IP oil works in the PNC:R refinery, and IP lubricant can be used to make PNC:R speed upgrades.  Diesel *should* work in the PNC:R thermo plant, but I didn't test that (JEI at least claims it works).
